doio.c

```diff
@@ -2967,6 +2971,9 @@ Perl_cando(pTHX_ Mode_t mode, bool effective, const Stat_t *statbufp)
      /* Atari stat() does pretty much the same thing. we set x_bit_set_in_stat
       * too so it will actually look into the files for magic numbers
       */
+# ifdef __MINGW32__
+    PERL_UNUSED_ARG(effective);
+# endif
```

Comment:
Does the `#ifdef __MINGW32__` guard make sense here? If `effective` is unused in this branch (`DOSISH`), we could just mark it as `PERL_UNUSED_ARG` unconditionally, even if other compilers don't warn about it (yet?).
Or am I missing something?

Reply:
> Does the <code>#ifdef __MINGW32__ </code> guard make sense here?

I don't think itserves a purpose, and I've now removed it.
I was bit thrown by the fact that I don't know what the "DOSISH" systems are ... and lost sight of the fact that it doesn't actually matter.

Reply:
> I was bit thrown by the fact that I don't know what the "DOSISH" systems are

At this point it's really only WinNT, but historically it also included Win95, DOS and OS/2